### PR TITLE
[Merged by Bors] - feat(Combinatorics/Quiver/ReflQuiver): reflexive quivers

### PR DESCRIPTION
**Pull request by digama0**
Reflexive quivers are an intermediate structure between categories and quivers, containing only an identity element. This PR constructs `ReflQuiver` as a class and `ReflQuiv` which is the category of reflexive quivers and reflexive prefunctors.

Co-Authored-By: Emily Riehl <eriehl@jhu.edu> and Pietro Monticone <38562595+pitmonticone@users.noreply.github.com>

---

* Unsolved: Should we refactor `Category` so that `Category extends ReflQuiver`?
* Prerequisite for #16784.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/Quiver/ReflQuiver.lean [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+/-!
+# Reflexive Quivers
+
+This module defines reflexive quivers. A reflexive quiver, or "refl quiver" for short, extends
+a quiver with a specified endoarrow on each term in its type of objects.
+
+We also introduce morphisms between reflexive quivers, called reflexive prefunctors or "refl
+prefunctors" for short.
+-/
```
Comment (semorrison) [inline Mathlib/Combinatorics/Quiver/ReflQuiver.lean:17]: Maybe some note saying that `Category` doesn't extend `ReflQuiver`, and why?

File: Mathlib/Combinatorics/Quiver/ReflQuiver.lean [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+-- Porting note: added during port.
+-- These lemmas can not be `@[simp]` because after `whnfR` they have a variable on the LHS.
+-- Nevertheless they are sometimes useful when building functors.
+lemma mk_obj {V W : Type*} [ReflQuiver V] [ReflQuiver W] {obj : V → W} {map} {X : V} :
+    (Prefunctor.mk obj map).obj X = obj X := rfl
+
+lemma mk_map {V W : Type*} [ReflQuiver V] [ReflQuiver W] {obj : V → W} {map} {X Y : V} {f : X ⟶ Y} :
+    (Prefunctor.mk obj map).map f = map f := rfl
+
+-- @[ext]
```
Comment (semorrison) [inline Mathlib/Combinatorics/Quiver/ReflQuiver.lean:54]:
```suggestion
-- @[ext]
```
Remove?

**Comment by semorrison** [comment]
bors d+

File: Mathlib/Combinatorics/Quiver/ReflQuiver.lean (hunk 1 above)
Comment (digama0) [reply Mathlib/Combinatorics/Quiver/ReflQuiver.lean:17]: As mentioned in the PR description, this is a TODO. It's pending zulip discussion or so, so there is not much in the way of "why" beyond it being a significant refactor which should go in a separate PR.

File: Mathlib/Combinatorics/Quiver/ReflQuiver.lean [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+/-- A morphism of reflexive quivers called a `ReflPrefunctor`. -/
+structure ReflPrefunctor (V : Type u₁) [ReflQuiver.{v₁} V] (W : Type u₂) [ReflQuiver.{v₂} W]
+    extends Prefunctor V W where
+  /-- A functor preserves identity morphisms. -/
+  map_id : ∀ X : V, map (𝟙rq X) = 𝟙rq (obj X) := by aesop_cat
+
+namespace ReflPrefunctor
+
+-- Porting note: added during port.
```
Comment (b-mehta) [inline Mathlib/Combinatorics/Quiver/ReflQuiver.lean:47]:
```suggestion
```

**Comment by sgouezel** [comment]
bors r-
Can you merge master, fix the build, and send it back to bors?
bors d+

**Comment by emilyriehl** [comment]
I don't get why this build failed @digama0 @sgouezel @kim-em. I just checked it out and it succeeds for me.

**Comment by kim-em** [comment]
I would suggest `git merge origin/master` and push that.

**Comment by emilyriehl** [comment]
Thanks @kim-em. The issue was actually with the new "'" linter, which I hadn't encountered before. But that's resolved now.

**Pull request closed without merging**